quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -571,46 +571,96 @@ impl Metastore for PostgresqlMetastore {
         Ok(())
     }
 
-    #[instrument(skip(self, split_metadata), fields(index_id=index_id, split_id=split_metadata.split_id))]
-    async fn stage_split(
+    #[instrument(skip(self, split_metadata_list))]
+    async fn stage_splits(
         &self,
         index_id: &str,
-        split_metadata: SplitMetadata,
+        split_metadata_list: Vec<SplitMetadata>,
     ) -> MetastoreResult<()> {
-        let split_metadata_json = serde_json::to_string(&split_metadata).map_err(|error| {
-            MetastoreError::JsonSerializeError {
-                struct_name: "SplitMetadata".to_string(),
-                message: error.to_string(),
+        let mut split_ids = Vec::with_capacity(split_metadata_list.len());
+        let mut time_range_start_list = Vec::with_capacity(split_metadata_list.len());
+        let mut time_range_end_list = Vec::with_capacity(split_metadata_list.len());
+        let mut tags_list = Vec::with_capacity(split_metadata_list.len());
+        let mut split_metadata_json_list = Vec::with_capacity(split_metadata_list.len());
+        let mut delete_opstamps = Vec::with_capacity(split_metadata_list.len());
+
+        for split_metadata in split_metadata_list {
+            let split_metadata_json = serde_json::to_string(&split_metadata).map_err(|error| {
+                MetastoreError::JsonSerializeError {
+                    struct_name: "SplitMetadata".to_string(),
+                    message: error.to_string(),
+                }
+            })?;
+            let time_range_start = split_metadata
+                .time_range
+                .as_ref()
+                .map(|range| *range.start());
+            let time_range_end = split_metadata.time_range.map(|range| *range.end());
+            let tags: Vec<String> = split_metadata.tags.into_iter().collect();
+
+            split_ids.push(split_metadata.split_id);
+            time_range_start_list.push(time_range_start);
+            time_range_end_list.push(time_range_end);
+            tags_list.push(sqlx::types::Json(tags));
+            split_metadata_json_list.push(split_metadata_json);
+            delete_opstamps.push(split_metadata.delete_opstamp as i64);
```

Comment:
Can you reorder the lines to make more sense?

Reply:
I've shuffled them around so that they're grouped rather than separated.